test/test_optim.py

```diff
@@ -431,6 +431,21 @@ def test_adam(self):
             with self.assertRaisesRegex(ValueError, "Invalid weight_decay value: -1"):
                 optimizer(None, lr=1e-2, weight_decay=-1)
 
+    # Test whether variance parameter is always real
+    def test_complex_adam_variance(self):
+        complex_param = torch.randn(5, 5, dtype=torch.complex64, requires_grad=True)
+        target = torch.randn(5, 5, dtype=torch.complex64)
+        optimizer = optim.Adam([complex_param], lr=0.001)
+
+        for i in range(200):
```

Comment:
Why don't you use `_test_basic_cases` here?

Reply:
The main reason i didn't include this test to `_test_basic_cases` , is that not all optimizers have variance term, and this issue is specific to certain subset of optimizers. That's why having a separate test sounded better idea to me. Do you think you agree with me ?

Reply:
If we're moving it to 20 iterations, then yes that sounds good.